nomad/deployment_endpoint.go

```diff
@@ -491,8 +504,14 @@ func (d *Deployment) List(args *structs.DeploymentListRequest, reply *structs.De
 
 			tokenizer := paginator.NewStructsTokenizer(iter, opts)
 
+			filters := []paginator.Filter{
+				paginator.NamespaceFilter{
+					AllowableNamespaces: allowableNamespaces,
+				},
+			}
+
 			var deploys []*structs.Deployment
-			paginator, err := paginator.NewPaginator(iter, tokenizer, nil, args.QueryOptions,
+			paginator, err := paginator.NewPaginator(iter, tokenizer, filters, args.QueryOptions,
```

Comment:
While we are in this area, could we change the name of the `paginator` variable as it clashes with an import.